Assets/Editor/ModelInspector.cs

```diff
@@ -247,9 +288,34 @@ private void SetDisplayMode(object data)
                 case DisplayMode.Tangent:
                     OnDropDownAction(m_Settings.meshMultiPreviewMaterial, 3, false);
                     break;
+                case DisplayMode.Blendshapes:
+                    OnDropDownAction(m_Settings.shadedPreviewMaterial, 0, false);
+                    GetSkinnedMeshRender();
+                    SetSingleBlendShapeWeight();
+                    break;
             }
         }
         
+        private void SetBlendshape(object data)
+        {
+            int popupIndex = (int)data;
+            if (popupIndex < 0 || popupIndex >= m_BlendShapes.Count)
+                return;
+
+            m_Settings.activeBlendshape = popupIndex;
+            
+            SetSingleBlendShapeWeight();
+        }
+        
+        void SetSingleBlendShapeWeight()
```

Comment:
I think this can result in null exceptions in case SkinnedMeshRenderer was not found -- e.g. you happen to have just a Mesh asset (not part of some FBX), and there's no game object or SkinnedMeshRenderer anywhere.

Reply:
Maybe it would be more useful to not try to use any existing SkinnedMeshRenderer, but rather create one from code in the preview. I'm not sure if for example it just uses default bind pose if you don't setup `.bones` on it (if it does, great! easy). Otherwise probably would also need to create the fake bone game objects and make their transforms match the default mesh bind poses.

I wonder how e.g. animation preview code does it...